### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by nikitavbv**

### CODE REVIEW

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+        }
+
+        private static void ValidateRenderOptions(Scene scene, RenderOptions renderOptions)
+        {
+            if (FindCameraById(scene, renderOptions.CameraId) == null)
+            {
+                throw new SceneIOException("RenderOptions references camera with id = " + renderOptions.CameraId + " which is not present on the scene.");
+            }
+
+            if (renderOptions.Width < 0)
```
Comment (Brookoko) [inline csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: 0 should be invalid too

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+            {
+                throw new SceneIOException("RenderOptions width cannot be negative");
+            }
+
+            if (renderOptions.Height < 0)
```
Comment (Brookoko) [inline csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Same as above

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+            }
+
+            if (sceneObject.Material != null)
+            {
+                ValidateMaterial(sceneObject.Material);
+            }
+
+            if (sceneObject.Sphere != null)
+            {
+                if (sceneObject.Sphere.Radius < 0)
```
Comment (Brookoko) [inline csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Same as above

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+
+                if (HasScale(sceneObject.Transform))
+                {
+                    throw new SceneIOException("Scale cannot be applied to Cube, use size vector instead.");
+                }
+            }
+
+            if (sceneObject.Disk != null)
+            {
+                if (sceneObject.Disk.Radius < 0)
```
Comment (Brookoko) [inline csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Same as above

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs (hunk 1 above)
Comment (nikitavbv) [reply csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Fixed, thanks.

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs (hunk 2 above)
Comment (nikitavbv) [reply csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Fixed.

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs (hunk 3 above)
Comment (nikitavbv) [reply csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Fixed.

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs (hunk 4 above)
Comment (nikitavbv) [reply csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Fixed.

File: csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs [hunk 5]
```diff
@@ -0,0 +1,581 @@
...
+                    }
+                }
+                
+                ValidateColor(material.LambertReflection.Color);
+            }
+        }
+
+        private static void ValidateColor(Color color)
+        {
+            if (color.R <= 0 || color.G <= 0 || color.B <= 0)
```
Comment (Brookoko) [inline csharp/SceneFormat/SceneFormat/ScenePostProcessing.cs]: Invalid validation. Color (0, 0, 0) is valid color

**Review by Brookoko — APPROVED** [review]

**Pull request merged**